core/lib/src/response/status.rs

```diff
@@ -211,6 +211,52 @@ impl<'r, R: Responder<'r>> Responder<'r> for Accepted<R> {
     }
 }
 
+/// Sets the status of the response to 204 (No Content).
+///
+/// If a responder is supplied, the remainder of the response is delegated to
+/// it. The body will be empty even if the responder sets a non-empty body.
+///
+#[derive(Debug, Clone, PartialEq)]
+pub struct NoContent<R>(pub Option<R>);
```

Comment:
Let's simplify this to `struct NoContent;` to avoid allowing a body that won't even be used.

Reply:
I think you are right. My logic was that Responders might be used to set some http response headers, however I don't think this would be so common. Since these wrappers are there for as just simple tools, it's probably not a big deal to say that anyone who wants something like setting an additional header for `No Content` might write their own responders or wrappers to do so anyway.

Reply:
(The same is true for ResetContent's responder)

Reply:
They could also still wrap in the opposite direction if they need that, assuming the Responder that adds the headers really only adds headers: `MyHeaders<NoContent>`.